### PR TITLE
GCS direct uploads

### PR DESCRIPTION
**Pull request by fiedosiukr**
## Changes
- Added [Dropzone.js](https://www.dropzone.dev/js/) to handle direct uploads to GCS
- Added conditional template rendering for different storage types
- Created an endpoint for generating signed urls
- Improved some of the JS code

### CODE REVIEW

File: FILE-STORAGE.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- LOCAL (files are stored on the same filesystem as the hosting server)
+- GCP
+
+## Local Filesystem
+
+- Static files are stored in the `STATIC_ROOT` folder. Directly served by nginx.
+- User-uploaded media files are stored in the `MEDIA_ROOT` folder. Authorization is done by Django views and then an `X-Accel-Redirect` to nginx if approved.
+- When projects get published, if it is a public project, files get moved to a `published_projects` subdirectory within `STATIC_ROOT`. Otherwise they get moved into a `published_projects` subdirectory within `MEDIA_ROOT`.
+
+## GCS File Storage
```
Comment (grzegorz-wcislo) [inline FILE-STORAGE.md]: Could you add info regarding CORS configuration for the media bucket (required for direct uploads)

File: physionet-django/console/views.py [hunk 2]
```diff
@@ -411,7 +411,7 @@ def copyedit_submission(request, project_slug, *args, **kwargs):
     if 'subdir' not in vars():
         subdir = ''
 
-    authors, author_emails, storage_info, edit_logs, copyedit_logs, latest_version = project.info_card()
+    authors, author_emails, storage_info, edit_logs, copyedit_logs, latest_version = project.info_card(force_calculate=True)
```
Comment (grzegorz-wcislo) [inline physionet-django/console/views.py]: I'm not sure if we want to use force_calculate when using local file storage here

File: physionet-django/project/templates/project/edit_files_panel.html [hunk 3]
```diff
@@ -1,7 +1,20 @@
 {% load static %}
 
+{% block local_js_top %}
+<script src="{% static 'dropzone/dropzone.min.js' %}"></script>
+{% endblock %}
+
+{% block local_css %}
+<link rel="stylesheet" type="text/css" href="{% static 'dropzone/dropzone.min.css' %}">
```
Comment (grzegorz-wcislo) [inline physionet-django/project/templates/project/edit_files_panel.html:8]: What's the difference between `dropzone.min.css` and `basic.min.css`? Do we need both?

Comment (fiedosiukr) [reply physionet-django/project/templates/project/edit_files_panel.html:8]: It seems like having `dropzone.min.css` is sufficient.

**Pull request closed without merging**